flang/runtime/CMakeLists.txt

```diff
@@ -169,6 +169,7 @@ set(sources
   unit-map.cpp
   unit.cpp
   utf.cpp
+  ${FORTRAN_MODULE_OBJECTS}
```

Comment:
This breaks some of my builds.  I believe this won't work in general unless you make sure that the object file is built before `FortranRuntime` target tries to build, i.e. there has to be a target dependency in this CMake file.

I think you can try to reproduce it by deleting the module object file and running `make FortranRuntime` - it will fail with something like:
```
No rule to make target `tools/flang/tools/f18/iso_fortran_env_impl.o', needed by `lib/libFortranRuntime.a'.  Stop.
```

Reply:
On it.  Seems like Ninja understands the dependency, as it has a single build file that seems the generated object file in one place and defers the target in the runtime libraries until that object file has been built.  This does not seem to be the case for Makefile builds, because they are split across different Makefiles.

Reply:
Created https://github.com/llvm/llvm-project/pull/99737 to hopefully resolve this.  Can you please see if the PR resolves the issue?

And sorry for breaking your builds!

Reply:
Thank you, Michael.  No problem, it should fix my builds, and building `module_files` target before the complete build was a workaround for me.